lib/ansible/playbook/play_context.py

```diff
@@ -535,7 +535,8 @@ def detect_su_prompt(b_data):
             elif self.become_method == 'pbrun':
 
                 prompt='Password:'
-                becomecmd = '%s %s -u %s %s' % (exe, flags, self.become_user, success_cmd)
+                becomecmd = '%s -b %s -u %s %s' % (exe, flags, self.become_user, '%s -c %s' % (executable, success_cmd))
```

Comment:
we've had problems with using -b in previous iterations, you should be able to add it as a dynamic flag if needed, hardcoding it is a bad idea

we had to take out the shell also because of many installations having this restricted